notebooks/Benchmark_notebook.ipynb

```diff
@@ -197,23 +217,26 @@
     "    \"\"\"Evaluates given integrator and returns result and runtime\n",
     "    \"\"\"\n",
     "    start = perf_counter()\n",
-    "    result = integrator(fn,dim,N,domain)\n",
+    "    result, grid_runtime, fn_runtime = integrator(fn,dim,N,domain)\n",
     "    time = perf_counter() - start\n",
-    "    return result,time\n",
+    "    return result, grid_runtime, fn_runtime, time -(grid_runtime + fn_runtime)\n",
     "\n",
     "def benchmark(integrator,fn,dim,N,domain, iterations=10):\n",
     "    \"\"\" Evaluates given Integrator once as warmup, then #iterations times\n",
     "    \"\"\"\n",
-    "    results,times = [],[]\n",
+    "    results = []\n",
+    "    grid_runtime, fn_runtime, integration_time = 0, 0, 0\n",
     "    #Do three calls as warmup (to allow CPU / GPU to use caching and increase frequency)\n",
     "    for i in range(WARMUP_CALLS): \n",
     "        evaluate(integrator,fn,dim,N,domain)\n",
     "    #Do iteration benchmark calls\n",
     "    for it in range(iterations):\n",
```

Comment:
Okay, so here, I think, you need to set `runtime_f = 0` (warmup calls should not be part of the measured time of course)

Reply:
(before the loop I mean)